### ISSUE TITLE
Why my R,P,F are 0.0000?I can't find the answer from validate.

### ISSUE THREAD

**Issue by Xiyujiang0203** [huiqu18/WeaklySegPartialPoints#8]
> ray@215-workstation-gen2 ~/w/j/WeaklySegPartialPoints-master> sh run.sh                        (jxy_PIDnet) 
> /home/ray/workspace/jxy/WeaklySegPartialPoints-master/code_detection
> 
> # ------------------------- Options ------------------------- #
>              isTrain: True                               
>              dataset: MO                                 
>                ratio: 0.1                                
>                   r1: 11                                 
>                   r2: 22                                 
>       gaussian_sigma: 2.75                               
> 
> *************** model ***************
>                 name: ResUNet34                          
>           pretrained: True                               
>                out_c: 1                                  
> 
> *************** train ***************
>          random_seed: -1                                 
>             data_dir: ../data_for_train/MO               
>        root_save_dir: ../experiments/detection/MO/0.10_repeat=1
>           input_size: 224                                
>         train_epochs: 80                                 
>           batch_size: 16                                 
>      checkpoint_freq: 100                                
>                   lr: 0.001                              
>         weight_decay: 0.0001                             
>         log_interval: 30                                 
>              workers: 1                                  
>                 gpus: [0]                                
>              img_dir: ../data_for_train/MO/images        
>            label_dir: ../data_for_train/MO/labels_detect 
>               bg_dir: ../data_for_train/MO/labels_bg     
> 
> *************** transform ***************
> # -------------------------- End -------------------------- #
> 
> =============== initial training ===============
> => Preparing training samples
> /mnt/data/workspaces/jxy/WeaklySegPartialPoints-master/code_detection/prepare_data.py:67: UserWarning: ../data/MO/labels_point_0.10/Kidney_TCGA-HE-7128-01Z-00-DX1_label_point.png is a low contrast image
>   io.imsave('{:s}/{:s}'.format(new_label_point_dir, imgname), (label_partial_point > 0).astype(np.uint8) * 255)
> /mnt/data/workspaces/jxy/WeaklySegPartialPoints-master/code_detection/prepare_data.py:121: UserWarning: ../data/MO/patches/labels_detect/Breast_TCGA-A7-A13E-01Z-00-DX1_13_label_detect.png is a low contrast image
>   io.imsave('{:s}/{:s}_{:d}_{:s}.png'.format(save_dir, name[:-len(post_fix)-1], k, post_fix), seg_imgs[k])
> /mnt/data/workspaces/jxy/WeaklySegPartialPoints-master/code_detection/prepare_data.py:121: UserWarning: ../data/MO/patches/labels_detect/Breast_TCGA-A7-A13F-01Z-00-DX1_2_label_detect.png is a low contrast image
>   io.imsave('{:s}/{:s}_{:d}_{:s}.png'.format(save_dir, name[:-len(post_fix)-1], k, post_fix), seg_imgs[k])
> /mnt/data/workspaces/jxy/WeaklySegPartialPoints-master/code_detection/prepare_data.py:121: UserWarning: ../data/MO/patches/labels_detect/Breast_TCGA-A7-A13F-01Z-00-DX1_3_label_detect.png is a low contrast image
>   io.imsave('{:s}/{:s}_{:d}_{:s}.png'.format(save_dir, name[:-len(post_fix)-1], k, post_fix), seg_imgs[k])
> /mnt/data/workspaces/jxy/WeaklySegPartialPoints-master/code_detection/prepare_data.py:121: UserWarning: ../data/MO/patches/labels_detect/Breast_TCGA-A7-A13F-01Z-00-DX1_10_label_detect.png is a low contrast image
>   io.imsave('{:s}/{:s}_{:d}_{:s}.png'.format(save_dir, name[:-len(post_fix)-1], k, post_fix), seg_imgs[k])
> /mnt/data/workspaces/jxy/WeaklySegPartialPoints-master/code_detection/prepare_data.py:121: UserWarning: ../data/MO/patches/labels_detect/Kidney_TCGA-B0-5711-01Z-00-DX1_12_label_detect.png is a low contrast image
>   io.imsave('{:s}/{:s}_{:d}_{:s}.png'.format(save_dir, name[:-len(post_fix)-1], k, post_fix), seg_imgs[k])
> /mnt/data/workspaces/jxy/WeaklySegPartialPoints-master/code_detection/prepare_data.py:121: UserWarning: ../data/MO/patches/labels_detect/Kidney_TCGA-HE-7128-01Z-00-DX1_2_label_detect.png is a low contrast image
>   io.imsave('{:s}/{:s}_{:d}_{:s}.png'.format(save_dir, name[:-len(post_fix)-1], k, post_fix), seg_imgs[k])
> /mnt/data/workspaces/jxy/WeaklySegPartialPoints-master/code_detection/prepare_data.py:121: UserWarning: ../data/MO/patches/labels_detect/Liver_TCGA-38-6178-01Z-00-DX1_8_label_detect.png is a low contrast image
>   io.imsave('{:s}/{:s}_{:d}_{:s}.png'.format(save_dir, name[:-len(post_fix)-1], k, post_fix), seg_imgs[k])
> /mnt/data/workspaces/jxy/WeaklySegPartialPoints-master/code_detection/prepare_data.py:121: UserWarning: ../data/MO/patches/labels_detect/Liver_TCGA-38-6178-01Z-00-DX1_9_label_detect.png is a low contrast image
>   io.imsave('{:s}/{:s}_{:d}_{:s}.png'.format(save_dir, name[:-len(post_fix)-1], k, post_fix), seg_imgs[k])
> /mnt/data/workspaces/jxy/WeaklySegPartialPoints-master/code_detection/prepare_data.py:121: UserWarning: ../data/MO/patches/labels_detect/Liver_TCGA-38-6178-01Z-00-DX1_11_label_detect.png is a low contrast image
>   io.imsave('{:s}/{:s}_{:d}_{:s}.png'.format(save_dir, name[:-len(post_fix)-1], k, post_fix), seg_imgs[k])
> /mnt/data/workspaces/jxy/WeaklySegPartialPoints-master/code_detection/prepare_data.py:121: UserWarning: ../data/MO/patches/labels_detect/Liver_TCGA-49-4488-01Z-00-DX1_10_label_detect.png is a low contrast image
>   io.imsave('{:s}/{:s}_{:d}_{:s}.png'.format(save_dir, name[:-len(post_fix)-1], k, post_fix), seg_imgs[k])
> /mnt/data/workspaces/jxy/WeaklySegPartialPoints-master/code_detection/prepare_data.py:121: UserWarning: ../data/MO/patches/labels_detect/Prostate_TCGA-G9-6336-01Z-00-DX1_6_label_detect.png is a low contrast image
>   io.imsave('{:s}/{:s}_{:d}_{:s}.png'.format(save_dir, name[:-len(post_fix)-1], k, post_fix), seg_imgs[k])
> /mnt/data/workspaces/jxy/WeaklySegPartialPoints-master/code_detection/prepare_data.py:121: UserWarning: ../data/MO/patches/labels_detect/Prostate_TCGA-G9-6336-01Z-00-DX1_8_label_detect.png is a low contrast image
>   io.imsave('{:s}/{:s}_{:d}_{:s}.png'.format(save_dir, name[:-len(post_fix)-1], k, post_fix), seg_imgs[k])
> Computing the mean and standard deviation of training data...
> => Start training
> 07-03 04:34     ***** Training starts *****
> 07-03 04:34     save directory: ../experiments/detection/MO/0.10_repeat=1/0
> /home/ray/micromamba/envs/jxy_PIDnet/lib/python3.8/site-packages/torchvision/models/_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
>   warnings.warn(
> /home/ray/micromamba/envs/jxy_PIDnet/lib/python3.8/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=ResNet34_Weights.IMAGENET1K_V1`. You can also use `weights=ResNet34_Weights.DEFAULT` to get the most up-to-date weights.
>   warnings.warn(msg)
> 07-03 04:34     => Initial learning rate: 0.001
> 07-03 04:34     => Batch size: 16
> 07-03 04:34     => Number of training iterations: 960
> 07-03 04:34     => Training epochs: 80
> 07-03 04:34     Epoch: [1/80]
> /mnt/data/workspaces/jxy/WeaklySegPartialPoints-master/code_detection/my_transforms.py:73: UserWarning: TypedStorage is deprecated. It will be removed in the future and UntypedStorage will be the only storage class. This should only matter to you if you are using storages directly.  To access UntypedStorage directly, use tensor.untyped_storage() instead of tensor.storage()
>   pic = torch.ByteTensor(torch.ByteStorage.from_buffer(img.tobytes()))
> 07-03 04:34             Iteration: [0/12]       Loss 0.9830
> 07-03 04:34             => Train Avg: Loss 0.6619
> /mnt/data/workspaces/jxy/WeaklySegPartialPoints-master/code_detection/my_transforms.py:73: UserWarning: TypedStorage is deprecated. It will be removed in the future and UntypedStorage will be the only storage class. This should only matter to you if you are using storages directly.  To access UntypedStorage directly, use tensor.untyped_storage() instead of tensor.storage()
>   pic = torch.ByteTensor(torch.ByteStorage.from_buffer(img.tobytes()))
> 07-03 04:34             => Val Avg:     Recall 0.0000   Prec 0.0000     F1 0.0000
> 07-03 04:34     Epoch: [2/80]
> 07-03 04:34             Iteration: [0/12]       Loss 0.5757
> 07-03 04:34             => Train Avg: Loss 0.5353
> 07-03 04:34             => Val Avg:     Recall 0.0000   Prec 0.0000     F1 0.0000
> 07-03 04:34     Epoch: [3/80]
> 07-03 04:34             Iteration: [0/12]       Loss 0.4927
> 07-03 04:34             => Train Avg: Loss 0.5363
> 07-03 04:35             => Val Avg:     Recall 0.0000   Prec 0.0000     F1 0.0000
> 07-03 04:35     Epoch: [4/80]
> 07-03 04:35             Iteration: [0/12]       Loss 0.5400
> 07-03 04:35             => Train Avg: Loss 0.5204
> 07-03 04:35             => Val Avg:     Recall 0.0000   Prec 0.0000     F1 0.0000
> 07-03 04:35     Epoch: [5/80]
> 07-03 04:35             Iteration: [0/12]       Loss 0.4492
> 07-03 04:35             => Train Avg: Loss 0.5098
> 07-03 04:35             => Val Avg:     Recall 0.0000   Prec 0.0000     F1 0.0000
> 07-03 04:35     Epoch: [6/80]
> 07-03 04:35             Iteration: [0/12]       Loss 0.5218
> 07-03 04:35             => Train Avg: Loss 0.5037
> 07-03 04:35             => Val Avg:     Recall 0.0000   Prec 0.0000     F1 0.0000
> 07-03 04:35     Epoch: [7/80]
> 07-03 04:35             Iteration: [0/12]       Loss 0.5587
> 07-03 04:36             => Train Avg: Loss 0.5074
> 07-03 04:36             => Val Avg:     Recall 0.0000   Prec 0.0000     F1 0.0000
> 07-03 04:36     Epoch: [8/80]
> 07-03 04:36             Iteration: [0/12]       Loss 0.5705
> 07-03 04:36             => Train Avg: Loss 0.4995
> 07-03 04:36             => Val Avg:     Recall 0.0000   Prec 0.0000     F1 0.0000
> 07-03 04:36     Epoch: [9/80]
> 07-03 04:36             Iteration: [0/12]       Loss 0.4404
> 07-03 04:36             => Train Avg: Loss 0.4934
> 07-03 04:36             => Val Avg:     Recall 0.0000   Prec 0.0000     F1 0.0000
> 07-03 04:36     Epoch: [10/80]
> 07-03 04:36             Iteration: [0/12]       Loss 0.5103
> 07-03 04:36             => Train Avg: Loss 0.5059
> 07-03 04:36             => Val Avg:     Recall 0.0000   Prec 0.0000     F1 0.0000
> 07-03 04:36     Epoch: [11/80]
> 07-03 04:36             Iteration: [0/12]       Loss 0.5096
> 07-03 04:37             => Train Avg: Loss 0.4857
> 07-03 04:37             => Val Avg:     Recall 0.0000   Prec 0.0000     F1 0.0000
> 07-03 04:37     Epoch: [12/80]
> 07-03 04:37             Iteration: [0/12]       Loss 0.5135
> 07-03 04:37             => Train Avg: Loss 0.4894
> 07-03 04:37             => Val Avg:     Recall 0.0000   Prec 0.0000     F1 0.0000
> 07-03 04:37     Epoch: [13/80]
> 07-03 04:37             Iteration: [0/12]       Loss 0.5608
> 07-03 04:37             => Train Avg: Loss 0.4891
> 07-03 04:37             => Val Avg:     Recall 0.0000   Prec 0.0000     F1 0.0000
> 07-03 04:37     Epoch: [14/80]
> 07-03 04:37             Iteration: [0/12]       Loss 0.4589
> 07-03 04:37             => Train Avg: Loss 0.4873
> 07-03 04:37             => Val Avg:     Recall 0.0000   Prec 0.0000     F1 0.0000
> 07-03 04:37     Epoch: [15/80]
> 07-03 04:37             Iteration: [0/12]       Loss 0.4721
> 07-03 04:38             => Train Avg: Loss 0.4782
> 07-03 04:38             => Val Avg:     Recall 0.0000   Prec 0.0000     F1 0.0000
> 07-03 04:38     Epoch: [16/80]
> 07-03 04:38             Iteration: [0/12]       Loss 0.4956
> 07-03 04:38             => Train Avg: Loss 0.4973
> 07-03 04:38             => Val Avg:     Recall 0.0000   Prec 0.0000     F1 0.0000
